Sources/RichTextKit/Coordinator/RichTextCoordinator.swift

```diff
@@ -200,7 +200,20 @@ extension RichTextCoordinator {
                 richTextContext.setColor(color, for: $0)
             }
         }
-
+        
+        // IMPORTANT:
+        // When we set custom RichTextAttributes, those are available only for parsing.
+        // That means when UITextView renders the attributed text, it automatically removes
+        // all undocumented attributes from itself and keep only the documented ones
+        // (In customLinks case - `.link` and `.foregroundColor`
+        // This is probably hack because it intervenes with other links (mentions implementation in future)
+        if let link = textView.currentRichTextAttributes[.link] as? String,
+           let url = URL(string: link){
+            richTextContext.setLink(url)
+        } else {
+            richTextContext.setLink(nil)
+        }
```

Comment:
This whole condition causes the funky behaviour when setting link and then selecting text together with link and text without attributes...

 The last issue I need to solve is that when I set a link, then select the text with link from the start and continue the selection to next characters, it sets the link for those next characters as well.. 
 
 This feels like there should be some another flag or check whether the current range is all link. I tried this with naive approach using `allSatisfy` on the selected text but I probably got it a bit wrong. Just in case if you have any idea if the logic could be handled a bit differently. cc @msrutek-paylocity

Reply:
I think I shouldn't set the link in here, rather just some flag, because this is responsible for telling the button to highlight or be disabled...

Reply:
Actually @danielsaidi I figured out that the logic for setting those attributes needs to be tweaked a bit. 

1. Setting the attributes occurs every time, even if `textViewDidChangeSelection` or `textViewDidChange` - Is this on purpose? 
2. With setting currentRichTextStyle, it appears we always set `typingAttributes` to textView even if we do not need to. We should probably change this and handle in a better way 
3. I am working on those improvements, will deliver those together with this PR probably or before this so I can continue with this.

Reply:
@DominikBucher12 Ok, sounds great! Just let me know when you want me to test anything. Some things were added to make certain operations work on all platforms, but they may also be outdated. It would be great if we can remove any not needed updates, since performance is currently an issue.

Reply:
Plot twist. The funny behaviour was caused by the Custom LayoutManager as I pointed out in here: https://github.com/danielsaidi/RichTextKit/pull/92#discussion_r1422310576 with the link :) Will add some tests, behaviour is now cool.

Reply:
This is fixed right now but will be tweaked with different handling of setting attributes (`userActionPublisher`) In order to support all attributes.